core/src/datayoga_core/blocks/files/read_csv/block.py

```diff
@@ -24,12 +26,29 @@ def init(self, context: Optional[Context] = None):
 
         logger.debug(f"file: {self.file}")
         self.batch_size = self.properties.get("batch_size", 1000)
+        self.fields = self.properties.get("fields")
+        self.skip = self.properties.get("skip", 0)
+        self.delimiter = self.properties.get("delimiter", ",")
+        self.quotechar = self.properties.get("quotechar", "\"")
 
     async def produce(self) -> AsyncGenerator[List[Message], None]:
         logger.debug("Reading CSV")
 
         with open(self.file, "r") as read_obj:
-            records = list(DictReader(read_obj))
+            reader = DictReader(read_obj, fieldnames=self.fields, delimiter=self.delimiter, quotechar=self.quotechar)
+            counter = iter(count())
```

Comment:
shouldn't this be a general util for producing message IDs?

Reply:
It needs to be discussed. I thought about uuid for these cases... Let's decide it later.